### PR TITLE
standardize terminology in Components docs

### PR DESCRIPTION
**Pull request by tdykstra**
partially addresses #2685

### CODE REVIEW

File: docs/standard/components.md [hunk 1]
```diff
@@ -14,37 +14,36 @@ ms.assetid: 2e38e9d9-8284-46ee-a15f-199adc4f26f4
...
-## .NET runtimes
+## .NET implementations
 
-There are 3 primary .NET runtimes which Microsoft actively develops and maintains: .NET Core, .NET Framework, and Mono for Xamarin.
+There are 3 primary .NET implementations which Microsoft actively develops and maintains: .NET Core, .NET Framework, and Mono for Xamarin.
 
 ### .NET Core
 
-.NET Core is a cross-platform runtime optimized for server workloads.  It implements the .NET Standard, which means that any code that targets the .NET Standard can run on .NET Core.  It is the runtime used by ASP.NET Core and the Universal Windows Platform (UWP).  It is modern, efficient, and designed to handle server and cloud workloads at scale.
+.NET Core is a cross-platform runtime optimized for server workloads.  It implements the .NET Standard, which means that any code that targets the .NET Standard can run on .NET Core.  It is the .NET implementation used by ASP.NET Core and the Universal Windows Platform (UWP).  It is modern, efficient, and designed to handle server and cloud workloads at scale.
```
Comment (GuardRex) [inline docs/standard/components.md]: The constructions that use an "it" pronoun can be a bit slower to pick up by the reader. This is a really fine example imo ...

Is the "It" of the 2nd sentence referring to ".NET Core" or to the "runtime"?

... then I see a lot of "it"s in this para. I recommend refactoring to get rid of at least a few of them.

Contraction: It is :point_right: It's (occurs twice in the para)

.NET Core isn't just a "cross-platform runtime" as the opening sentence states. I suggest "includes" in there ...

> .NET Core includes a cross-platform runtime optimized for server workloads.

> ... any code that targets the .NET Standard can run on .NET Core ...

"any code" ... I hope so, but I have a bad feel'in about this, and I guess it's not good if it were to say "most code." Idk if I'm right/wrong about it today, but I'm just worried that ye 'olde `NotImplementedException` will pop up from time-to-time, so the statement is more of a goal than a hardcore fact if that's still the case.

Two-spaces between sentences here.

File: docs/standard/components.md [hunk 2]
```diff
@@ -14,37 +14,36 @@ ms.assetid: 2e38e9d9-8284-46ee-a15f-199adc4f26f4
...
 
 To learn more about .NET Core, see the [.NET Core Guide](../core/index.md).
 
 ### .NET Framework
 
-.NET Framework is the historical .NET runtime that has existed since 2002.  It is the same .NET Framework existing .NET developers have always used.  It implements the .NET Standard, which means that any code that targets the .NET Standard can run on the .NET Framework.  It contains additional Windows-specific APIs, such as APIs for Windows desktop development with Windows Forms and WPF.  .NET Framework is optimized for building Windows desktop applications.
+.NET Framework is the original .NET implementation that has existed since 2002.  It is the same .NET Framework that existing .NET developers have always used.  It implements the .NET Standard, which means that any code that targets the .NET Standard can run on the .NET Framework.  It contains additional Windows-specific APIs, such as APIs for Windows desktop development with Windows Forms and WPF. The .NET Framework is optimized for building Windows desktop applications.
```
Comment (GuardRex) [inline docs/standard/components.md]: > The .NET Framework ...

It is :point_right: It's

For ...

> It implements the .NET Standard

... since that hasn't always been the case, I wonder if you want to qualify it with something like "Now, ..." or "Since \<year>, ..." to indicate it's a relatively recent feature.

More "it" subjects here. I recommend upgrading a few of these to concrete subjects.

> ... any code that targets the .NET Standard ...

"any" ... Again, I think there were (historical) examples of this not being true.

Two-spaces between sentences here.

File: docs/standard/components.md [hunk 3]
```diff
@@ -54,27 +53,34 @@ Mono is the runtime used by Xamarin apps.  It implements the .NET Standard, whic
...
+A runtime is the execution environment for a managed program. The OS is part of the runtime environment but is not part of the .NET runtime. Here are some examples of .NET runtimes:
+ 
+ - Common Language Runtime (CLR) for .NET Framework
+ - Core Common Language Runtime (CoreCLR) for .NET Core
+ - .NET Native for Universal Windows Platform 
+
 ## .NET tooling and common infrastructure
 
-Tooling for .NET is also common across each implementation of .NET.  These include, but are not limited to:
+Some tooling for .NET is common across all implementations of .NET.  These include, but are not limited to:
```
Comment (GuardRex) [inline docs/standard/components.md]: Two-spaces before "These"

File: docs/standard/components.md [hunk 4]
```diff
@@ -54,27 +53,34 @@ Mono is the runtime used by Xamarin apps.  It implements the .NET Standard, whic
...
 
-* The .NET languages and their compilers
-* Runtime components, such as the JIT and Garbage Collector
-* .NET project system (sometimes known as "csproj", "vbproj", or "fsproj")
-* MSBuild, the build engine used to build projects
-* NuGet, Microsoft's package manager for .NET
-* The .NET Core CLI, a cross-platform command-line interface for building .NET projects
-* Open Source build orchestration tools, such as CAKE and FAKE
+- The .NET languages and their compilers
+- .NET project system (sometimes known as "csproj", "vbproj", or "fsproj")
```
Comment (GuardRex) [inline docs/standard/components.md]: Commas inside of quotes, but I admit doing something different in cases like this because these are file extensions (types), too, and there are similar cases where I didn't want there to be any confusion about punctuation marks and technical terms. I've made good use of emphasis ... just a suggestion ...

> .NET project system (sometimes known as \*csproj\*, \*vbproj\*, or \*fsproj\*)

.NET project system (sometimes known as *csproj*, *vbproj*, or *fsproj*)

File: docs/standard/components.md [hunk 5]
```diff
@@ -54,27 +53,34 @@ Mono is the runtime used by Xamarin apps.  It implements the .NET Standard, whic
...
+- MSBuild, the build engine used to build projects
+- NuGet, Microsoft's package manager for .NET
+- Open Source build orchestration tools, such as CAKE and FAKE
```
Comment (GuardRex) [inline docs/standard/components.md]: There are some opportunities here to cross-link if u want to ...

MSBuild :point_right: \[MSBuild](/visualstudio/msbuild/msbuild)
NuGet :point_right: \[NuGet](/nuget/)
CAKE :point_right: \[CAKE](http://cakebuild.net/)
FAKE :point_right: \[FAKE](https://fake.build/)

File: docs/standard/components.md [hunk 6]
```diff
@@ -54,27 +53,34 @@ Mono is the runtime used by Xamarin apps.  It implements the .NET Standard, whic
...
 
-The main takeaway here should be that there is a vast amount of tooling and infrastructure which is common for any "flavor" of .NET you choose to build your apps with.
+The main takeaway here should be that there is a vast amount of tooling and infrastructure which is common for any implementation of .NET you choose to build your apps with.
 
 ## Next Steps
```
Comment (GuardRex) [inline docs/standard/components.md]: They're going with "See also" for the links section.

File: docs/standard/components.md [hunk 7]
```diff
@@ -54,27 +53,34 @@ Mono is the runtime used by Xamarin apps.  It implements the .NET Standard, whic
...
-* The .NET Core CLI, a cross-platform command-line interface for building .NET projects
-* Open Source build orchestration tools, such as CAKE and FAKE
+- The .NET languages and their compilers
+- .NET project system (sometimes known as "csproj", "vbproj", or "fsproj")
+- MSBuild, the build engine used to build projects
+- NuGet, Microsoft's package manager for .NET
+- Open Source build orchestration tools, such as CAKE and FAKE
 
-The main takeaway here should be that there is a vast amount of tooling and infrastructure which is common for any "flavor" of .NET you choose to build your apps with.
+The main takeaway here should be that there is a vast amount of tooling and infrastructure which is common for any implementation of .NET you choose to build your apps with.
```
Comment (GuardRex) [inline docs/standard/components.md]: "here" is unnecessary. Suggest a definitive statement; the "takeaway here" is conversational but wordy. "a vast amount of X and Y" seems vague and awkward. "which" :point_right: "that" ... and the prep is avoidable if u like.

> You have access to an extensive set of tools and infrastructure components that are available for any implementation of .NET you employ to build your apps and libraries.

mmmmm .... ahhhhh lol

**Review by guardrex — CHANGES_REQUESTED** [review]
Very nice and mostly just nits. I have a few concerns in passing on some of the text that pre-dates your set of changes here.

**Review by guardrex — APPROVED** [review]

**Comment by GuardRex** [comment]
@tdykstra Looks great ... recommend tho the one-space between sentences.

**Comment by tdykstra** [comment]
@GuardRex Thank you for a thorough review as usual!  Will do on the sentence spacing.

**Comment by GuardRex** [comment]
@tdykstra Build looks good. In case you haven't seen the relative links breaking on the [review site](https://review.docs.microsoft.com/en-us/dotnet/standard/components?branch=pr-en-us-2753&view=netframework-4.7)* (i.e., MSBuild and NuGet links here), you don't need to sweat them, as they'll sort themselves out on the live site with the correct domain.

*External reviewers, please use the PR diff.

**Comment by tdykstra** [comment]
@mairaw @BillWagner After the changes to the plans for this doc in the issue, we could merge this PR as an intermediate step that fixes the most egregious errors, or just close it and start over.  Which would you like to do?

**Comment by GuardRex** [comment]
Side note @tdykstra (putting on my OSS community dude hat now): This is one of the best topics MS has *EVER* put online. Congrats on this! :tada: 🎈 :+1: 💓 :rocket:

**Comment by tdykstra** [comment]
@GuardRex  Are you referring to this components doc or to the glossary?

**Comment by GuardRex** [comment]
@tdykstra My bad ... I meant glossary. :smile:

**Comment by tdykstra** [comment]
@BillWagner @mairaw There's more discussion in the issue (#2685) about what to do with this doc, but this PR addresses the immediate need to correct confusing terminology.  If it will be a while before we address the larger issues with the document's focus, it might be best to merge this PR in the meantime.

File: docs/standard/components.md [hunk 8]
```diff
@@ -1,6 +1,6 @@
 ---
-title: .NET Architectural Components
-description: Learn about key .NET architectural components, such as the .NET Standard, .NET runtimes, and tooling.
+title: .NET Architectural Components | Microsoft Docs
+description: Learn about .NET architectural components such as the .NET Standard, .NET Base Class Libraries, .NET runtimes, and tooling.
 keywords: .NET, .NET Standard, .NET Standard, .NET Core, .NET Framework, Xamarin, MSBuild, C#, F#, VB, compilers
 author: cartermp
 ms.author: mairaw
```
Comment (mairaw) [inline docs/standard/components.md]: refresh ms.date too

File: docs/standard/components.md [hunk 9]
```diff
@@ -1,6 +1,6 @@
 ---
-title: .NET Architectural Components
-description: Learn about key .NET architectural components, such as the .NET Standard, .NET runtimes, and tooling.
+title: .NET Architectural Components | Microsoft Docs
```
Comment (mairaw) [inline docs/standard/components.md]: we've stripped Microsoft Docs everywhere

File: docs/standard/components.md [hunk 10]
```diff
@@ -14,67 +14,69 @@ ms.assetid: 2e38e9d9-8284-46ee-a15f-199adc4f26f4
...
-
-More formally, it is a specification of .NET APIs which make up a uniform set of contracts that you compile your code against.  These contracts have underlying implementations for each .NET runtime.  This enables portability across different .NET runtimes, making it so that your code can effectively "run everywhere".
+The .NET Standard is a set of APIs that are implemented by the Base Class Library of a .NET implementation. More formally, it's a specification of .NET APIs that make up a uniform set of contracts that you compile your code against. These contracts are implemented in each .NET implementation. This enables portability across different .NET implementations, making it so that your code can effectively run everywhere.
 
-The .NET Standard is also a build target. You can currently target .NET Standard 1.0-2.0. If your code targets a version of the .NET Standard, it is guaranteed to run on any .NET runtime which implements that version.
+The .NET Standard is also a target framework. If your code targets a version of the .NET Standard, it can run on any .NET implementation which supports that version of .NET Standard.
```
Comment (mairaw) [inline docs/standard/components.md]: which supports -> that supports

File: docs/standard/components.md [hunk 11]
```diff
@@ -14,67 +14,69 @@ ms.assetid: 2e38e9d9-8284-46ee-a15f-199adc4f26f4
...
-* The .NET Core CLI, a cross-platform command-line interface for building .NET projects
-* Open Source build orchestration tools, such as CAKE and FAKE
+You have access to an extensive set of tools and infrastructure components that work with every implementation of .NET. These include, but are not limited to the following:
 
-The main takeaway here should be that there is a vast amount of tooling and infrastructure which is common for any "flavor" of .NET you choose to build your apps with.
+- The .NET languages and their compilers
+- The .NET project system (based on *.csproj*, *.vbproj*, and *.fsproj* files)
+- [MSBuild](/visualstudio/msbuild/msbuild), the build engine used to build projects
+- [NuGet](/nuget/), Microsoft's package manager for .NET
+- Open Source build orchestration tools, such as [CAKE](http://cakebuild.net/) and [FAKE](https://fake.build/)
```
Comment (mairaw) [inline docs/standard/components.md]: Open Source -> Open-source

File: docs/standard/components.md [hunk 12]
```diff
@@ -14,67 +14,69 @@ ms.assetid: 2e38e9d9-8284-46ee-a15f-199adc4f26f4
...
+The .NET Standard is an API specification common to all implementations of .NET. Implementations of .NET include the .NET Framework, .NET Core, and Mono for Xamarin. Each implementation of .NET includes the following components:
 
-Here's a graphical overview of each of the previously mentioned components of .NET and how they fit.
+- One or more runtimes. Examples: CLR for .NET Framework, CoreCLR and CoreRT for .NET Core.
+- A class library that implements the .NET Standard and may implement additional APIs. Examples: .NET Framework Base Class Library, .NET Core Base Class Library.
+- Optionally, one or more application frameworks. Examples: ASP.NET, Windows Forms, and WPF are included in the .NET Framework.
+- Optionally, development tools. Some development tools are shared among multiple implementations.
 
-![All .NET Architectural Components Together](media/components.png)
```
Comment (mairaw) [inline docs/standard/components.md]: wasn't there a good replacement for this image?

File: docs/standard/components.md [hunk 13]
```diff
@@ -14,67 +14,69 @@ ms.assetid: 2e38e9d9-8284-46ee-a15f-199adc4f26f4
 
 # .NET Architectural Components
 
-.NET is made up of a number of key components.  It has a standard library, called the .NET Standard, which is a large set of APIs which runs everywhere.  This standard library is implemented by three .NET runtimes - .NET Framework, .NET Core, and Mono for Xamarin.  The .NET languages also run on any .NET runtime.  Additionally, there are tools on every platform which allow you to build projects.  These tools are the same regardless of your choice of runtime.
+The .NET Standard is an API specification common to all implementations of .NET. Implementations of .NET include the .NET Framework, .NET Core, and Mono for Xamarin. Each implementation of .NET includes the following components:
```
Comment (mairaw) [inline docs/standard/components.md]: Can we open up with something about .NET? Or maybe why would you care about this topic? Seems too direct to start with .NET Standard.

File: docs/standard/components.md [hunk 14]
```diff
@@ -14,67 +14,69 @@ ms.assetid: 2e38e9d9-8284-46ee-a15f-199adc4f26f4
...
 
-Here's a graphical overview of each of the previously mentioned components of .NET and how they fit.
+- One or more runtimes. Examples: CLR for .NET Framework, CoreCLR and CoreRT for .NET Core.
+- A class library that implements the .NET Standard and may implement additional APIs. Examples: .NET Framework Base Class Library, .NET Core Base Class Library.
+- Optionally, one or more application frameworks. Examples: ASP.NET, Windows Forms, and WPF are included in the .NET Framework.
```
Comment (mairaw) [inline docs/standard/components.md]: should we spell out WPF on first usage? Also, would it be helpful to link to the docs here?

File: docs/standard/components.md [hunk 15]
```diff
@@ -14,67 +14,69 @@ ms.assetid: 2e38e9d9-8284-46ee-a15f-199adc4f26f4
...
+- Optionally, development tools. Some development tools are shared among multiple implementations.
 
-![All .NET Architectural Components Together](media/components.png)
-
-What follows is a brief explanation of each of the key components shown above.  
+What follows is a brief explanation of the key components mentioned in this overview.
```
Comment (mairaw) [inline docs/standard/components.md]: if we add an introduction in the opening, then this can go away,

File: docs/standard/components.md [hunk 16]
```diff
@@ -14,67 +14,69 @@ ms.assetid: 2e38e9d9-8284-46ee-a15f-199adc4f26f4
 
 # .NET Architectural Components
```
Comment (mairaw) [inline docs/standard/components.md]: since you've fixed all headings to the right casing, can we fix here too?

**Review by mairaw — COMMENTED** [review]
This looks really good @tdykstra. Thanks. Left some comments.

File: docs/standard/components.md (hunk 12 above)
Comment (tdykstra) [inline docs/standard/components.md]: Scott Hanselman is working on one.

File: docs/standard/components.md [hunk 17]
```diff
@@ -1,80 +1,82 @@
...
+
+Each implementation of .NET includes the following components:
 
-There are 3 primary .NET runtimes which Microsoft actively develops and maintains: .NET Core, .NET Framework, and Mono for Xamarin.
+- One or more runtimes. Examples: CLR for .NET Framework, CoreCLR and CoreRT for .NET Core.
+- A class library that implements the .NET Standard and may implement additional APIs. Examples: .NET Framework Base Class Library, .NET Core Base Class Library.
+- Optionally, one or more application frameworks. Examples: [ASP.NET](https://www.asp.net/), [Windows Forms](../framework/winforms/windows-forms-overview.md), and [Windows Presentation Foundation (WPF)](../framework/wpf/index.md) are included in the .NET Framework.
+- Optionally, development tools. Some development tools are shared among multiple implementations.
+
+There are three primary .NET implementations that Microsoft actively develops and maintains: .NET Core, .NET Framework, and Mono for Xamarin.
```
Comment (terrajobst) [inline docs/standard/components.md]: I think the relationship of Mono & Xamarin is similar to .NET Framework & .NET Core in that Xamarin is a subset and has specific runtimes. /cc @mhutch

I assume we're trying to make the point here that we don't actively maintain Mono but only Mono for Xamarin?

File: docs/standard/components.md [hunk 18]
```diff
@@ -1,80 +1,82 @@
...
 
 ### .NET Core
 
-.NET Core is a cross-platform runtime optimized for server workloads.  It implements the .NET Standard, which means that any code that targets the .NET Standard can run on .NET Core.  It is the runtime used by ASP.NET Core and the Universal Windows Platform (UWP).  It is modern, efficient, and designed to handle server and cloud workloads at scale.
+.NET Core is a cross-platform implementation of .NET optimized for server workloads. It's modern, efficient, and designed to handle server and cloud workloads at scale. It implements the .NET Standard, so code that targets the .NET Standard can run on .NET Core. ASP.NET Core and the Universal Windows Platform (UWP) run on .NET Core.
```
Comment (terrajobst) [inline docs/standard/components.md]: We need to separate .NET Core from UWP. The code is shared but there also quite different. I'd make .NET for UWP anther top level implementation and a peer to .NET Core.

**Review by terrajobst — CHANGES_REQUESTED** [review]
Looks good overall, but we need to separate UWP from .NET Core.

File: docs/standard/components.md (hunk 17 above)
Comment (mhutch) [inline docs/standard/components.md]: I think the wording is a little confusing, especially as Xamarin.Android used to be called "Mono for Android".

MS _supports_ Mono as part of the Xamarin SDKs, which integrate a subset of Mono called the "mobile" framework. That said, we _actively develop and maintain_ the "full" Mono framework, which we use as the runtime for VS for Mac. Not sure how best to express that succinctly though 😃

Maybe we can just say "Mono" here?

/cc @migueldeicaza

**Comment by tdykstra** [comment]
@terrajobst addressed feedback, is this ready to merge?

**Comment by tdykstra** [comment]
@terrajobst addressed feedback, is this ready to merge?

File: docs/standard/components.md [hunk 19]
```diff
@@ -1,80 +1,88 @@
...
+A .NET app is developed for and runs in one or more *implementations of .NET*.  Implementations of .NET include the .NET Framework, .NET Core, and Mono for Xamarin. There is an API specification common to all implementations of .NET that is called the .NET Standard. This article gives a brief introduction to each of these concepts.
 
 ## .NET Standard
 
-The .NET Standard is a set of APIs which are implemented by a .NET runtime.
-
-More formally, it is a specification of .NET APIs which make up a uniform set of contracts that you compile your code against.  These contracts have underlying implementations for each .NET runtime.  This enables portability across different .NET runtimes, making it so that your code can effectively "run everywhere".
+The .NET Standard is a set of APIs that are implemented by the Base Class Library of a .NET implementation. More formally, it's a specification of .NET APIs that make up a uniform set of contracts that you compile your code against. These contracts are implemented in each .NET implementation. This enables portability across different .NET implementations, making it so that your code can effectively run everywhere.
```
Comment (rpetrusha) [inline docs/standard/components.md]: "making it so that your code can" --> "allowing your code to"

File: docs/standard/components.md [hunk 20]
```diff
@@ -1,80 +1,88 @@
...
 
-The .NET Standard is also a build target. You can currently target .NET Standard 1.0-2.0. If your code targets a version of the .NET Standard, it is guaranteed to run on any .NET runtime which implements that version.
+The .NET Standard is also a [target framework](glossary.md#target-framework). If your code targets a version of the .NET Standard, it can run on any .NET implementation which supports that version of .NET Standard.
```
Comment (rpetrusha) [inline docs/standard/components.md]: of .NET Standard --> of the .NET Standard

File: docs/standard/components.md [hunk 21]
```diff
@@ -1,80 +1,88 @@
...
+Mono is the .NET implementation used by Xamarin apps. It implements the .NET Standard, so code that targets the .NET Standard can run on Xamarin apps. It contains additional APIs for iOS, Android, Xamarin.Forms, and Xamarin.Mac. It's optimized for building mobile applications on iOS and Android.
 
 To learn more about Mono, see the [Mono documentation](http://www.mono-project.com/docs/).
 
-## .NET tooling and common infrastructure
+### Universal Windows Platform (UWP)
+
+UWP is an implementation of .NET that is used for building modern, touch-enabled Windows applications and software for the Internet of Things (IoT). It's designed to unify the different types of devices that you may want to target, including PCs, tablets, phablets, phones, and even the Xbox. UWP provides many services, such as a centralized app store, an execution environment (AppContainer), and a set of Windows APIs to use instead of Win32 (WinRT). Apps can be written in C++, C#, VB.NET, and JavaScript. When using C# and VB.NET, the .NET APIs are provided by .NET Core.
```
Comment (rpetrusha) [inline docs/standard/components.md]: that is --> that's

File: docs/standard/components.md [hunk 22]
```diff
@@ -1,80 +1,88 @@
...
 
-Tooling for .NET is also common across each implementation of .NET.  These include, but are not limited to:
+To learn more about UWP, see [Intro to the Universal Windows Platform](https://docs.microsoft.com/windows/uwp/get-started/universal-application-platform-guide).
+
+## .NET runtimes
+
+A runtime is the execution environment for a managed program. The OS is part of the runtime environment but is not part of the .NET runtime. Here are some examples of .NET runtimes:
+ 
+ - Common Language Runtime (CLR) for .NET Framework
```
Comment (rpetrusha) [inline docs/standard/components.md]: .NET Framework --> the .NET Framework

**Review by rpetrusha — APPROVED** [review]
I suggested a few editorial corrections, but otherwise, LGTM, @tdykstra.

**Review by terrajobst — APPROVED** [review]
Looks good to me now, but I think you'll want @mhutch's approval for the Mono side of the house.

**Review by cartermp — APPROVED** [review]

File: docs/standard/components.md [hunk 23]
```diff
@@ -1,80 +1,88 @@
 ---
-title: .NET Architectural Components
-description: Learn about key .NET architectural components, such as the .NET Standard, .NET runtimes, and tooling.
-keywords: .NET, .NET Standard, .NET Standard, .NET Core, .NET Framework, Xamarin, MSBuild, C#, F#, VB, compilers
+title: .NET architectural components
+description: Describes .NET architectural components such as the .NET Standard, .NET implementations, .NET runtimes, and tooling.
+keywords: .NET, .NET Standard, .NET Core, .NET Framework, Xamarin, MSBuild, C#, F#, VB, compilers
```
Comment (mhutch) [inline docs/standard/components.md]: Can we add Mono here? :)

File: docs/standard/components.md [hunk 24]
```diff
@@ -1,80 +1,88 @@
...
 
 To learn more about the .NET Framework, see the [.NET Framework Guide](../framework/index.md).
 
-### Mono for Xamarin
+### Mono
 
-Mono is the runtime used by Xamarin apps.  It implements the .NET Standard, which means that any code that targets the .NET Standard can run on Xamarin apps.  It contains additional APIs for iOS, Android, Xamarin.Forms, and Xamarin.Mac.  It is optimized for building mobile applications on iOS and Android.
+Mono is the .NET implementation used by Xamarin apps. It implements the .NET Standard, so code that targets the .NET Standard can run on Xamarin apps. It contains additional APIs for iOS, Android, Xamarin.Forms, and Xamarin.Mac. It's optimized for building mobile applications on iOS and Android.
```
Comment (mhutch) [inline docs/standard/components.md]: The Xamarin.iOS/Xamarin.Android/Xamarin.Mac APIs are not part of Mono. I think that by the distinction that's being used here, Xamarin.iOS, Xamarin.Android and Xamarin.Mac are all ".NET implementations" that use the Mono runtime.

Xamarin.Forms isn't part of any of these - it's a NuGet that runs on UWP as well as Xamarin.iOS and Xamarin.Android.

Additionally, Mono has a desktop/server ".NET implementation" that is mostly compatible with .NET Framework, modulo Windows-specific APIs and a few other things. It doesn't have a formal name and is usually called just "Mono" or "Mono desktop framework". This implementation is what VS for Mac runs on.

File: docs/standard/components.md [hunk 25]
```diff
@@ -1,80 +1,88 @@
...
-Tooling for .NET is also common across each implementation of .NET.  These include, but are not limited to:
+To learn more about UWP, see [Intro to the Universal Windows Platform](https://docs.microsoft.com/windows/uwp/get-started/universal-application-platform-guide).
+
+## .NET runtimes
+
+A runtime is the execution environment for a managed program. The OS is part of the runtime environment but is not part of the .NET runtime. Here are some examples of .NET runtimes:
+ 
+ - Common Language Runtime (CLR) for the .NET Framework
+ - Core Common Language Runtime (CoreCLR) for .NET Core
+ - .NET Native for Universal Windows Platform
```
Comment (mhutch) [inline docs/standard/components.md]: - The Mono runtime for Xamarin.iOS, Xamarin.Android, Xamarin.Mac, and the Mono desktop framework.

File: docs/standard/components.md [hunk 26]
```diff
@@ -1,80 +1,88 @@
...
+# .NET architectural components
 
-.NET is made up of a number of key components.  It has a standard library, called the .NET Standard, which is a large set of APIs which runs everywhere.  This standard library is implemented by three .NET runtimes - .NET Framework, .NET Core, and Mono for Xamarin.  The .NET languages also run on any .NET runtime.  Additionally, there are tools on every platform which allow you to build projects.  These tools are the same regardless of your choice of runtime.
-
-Here's a graphical overview of each of the previously mentioned components of .NET and how they fit.
-
-![All .NET Architectural Components Together](media/components.png)
-
-What follows is a brief explanation of each of the key components shown above.  
+A .NET app is developed for and runs in one or more *implementations of .NET*.  Implementations of .NET include the .NET Framework, .NET Core, and Mono for Xamarin. There is an API specification common to all implementations of .NET that's called the .NET Standard. This article gives a brief introduction to each of these concepts.
```
Comment (mhutch) [inline docs/standard/components.md]: Not sure what "Mono for Xamarin" means here in the context of a list of implementations, see later comments.

**Review by mhutch — CHANGES_REQUESTED** [review]

File: docs/standard/components.md (hunk 24 above)
Comment (migueldeicaza) [inline docs/standard/components.md]: I would rephrase like this:

```
Mono is a .NET implementation that is mainly used when a small runtime is required.  It is the runtime that powers Xamarin applications on Android, Mac, iOS, tvOS and watchOS and is focused primarily on a small footprint.

It supports all of the currently published .NET Standards.

Historically Mono implemented the larger API of the .NET Framework and emulated some of the most
popular capabilities on Unix and is sometimes used to run .NET applications that rely on those capabilities
on Unix.

Mono is typically used with a just-in-time compiler, but also features a full static compiler (ahead-of-time compilation) that is used on platform like iOS.
```

Comment (tdykstra) [inline docs/standard/components.md]: So would it be accurate to refer to Mono as a collection of .NET implementations that share a common runtime?
